### PR TITLE
FOLIO: Modify notes field and add item_notes, issues, supplements and indexes

### PR DESCRIPTION
**Pull request by bbusenius**
**Changes in this request**:
- Updates the `noteFormatter` to only return notes that are public.
- Eliminates the deprecated `notes` field that was returning item notes and replaces it with the more descriptive `holdings_notes` field.
- Adds return values for `item_notes`, `issues`, `supplements` and `indexes`.

### CODE REVIEW

File: module/VuFind/src/VuFind/ILS/Driver/Folio.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/ILS/Driver/Folio.php]: I think it would be more clear to put the `?? []` fallback up above when initially defining `$holdingsStatements`, etc., rather than down here when you use the variables for the first time. I think there's a greater chance of PHP notices doing it this way.

File: module/VuFind/src/VuFind/ILS/Driver/Folio.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/ILS/Driver/Folio.php]: Is there always going to be a statement and a note? If not, it might be worth wrapping a `trim` around this to kill any superfluous whitespace.

File: module/VuFind/src/VuFind/ILS/Driver/Folio.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/ILS/Driver/Folio.php]: Is it worth wrapping an `array_filter` around this to strip out empty entries resulting from staff only notes?

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks for this -- looks like a much richer treatment of notes. I've just left a few small suggestions below. Once those are addressed, it might be good to see if @ihardy and/or @mis306lu have any further thoughts.

File: module/VuFind/src/VuFind/ILS/Driver/Folio.php
Comment (bbusenius) [reply module/VuFind/src/VuFind/ILS/Driver/Folio.php]: Yes, it's more concise too. This is done.

File: module/VuFind/src/VuFind/ILS/Driver/Folio.php
Comment (bbusenius) [reply module/VuFind/src/VuFind/ILS/Driver/Folio.php]: Great suggestion!

File: module/VuFind/src/VuFind/ILS/Driver/Folio.php
Comment (bbusenius) [reply module/VuFind/src/VuFind/ILS/Driver/Folio.php]: I don't see why not.

File: module/VuFind/src/VuFind/ILS/Driver/Folio.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/ILS/Driver/Folio.php]: Are `staffOnly` and `note` guaranteed to be set, or would it be safer to do something like this instead?

`return !($note->staffOnly ?? false) && !empty($note->note) ? $note->note : '';`

File: module/VuFind/src/VuFind/ILS/Driver/Folio.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/ILS/Driver/Folio.php]: I see you added an array_filter to the item notes filtering below; would it make sense to add one here as well for the holding notes?

**Comment by demiankatz** [comment]
Thanks, @bbusenius, this is looking great. I just had a couple small questions after a second review. Regardless of those, I think it would be fine for @ihardy and @mis306lu to take a look at this now, if they're able.

File: module/VuFind/src/VuFind/ILS/Driver/Folio.php
Comment (bbusenius) [reply module/VuFind/src/VuFind/ILS/Driver/Folio.php]: Done

File: module/VuFind/src/VuFind/ILS/Driver/Folio.php
Comment (bbusenius) [reply module/VuFind/src/VuFind/ILS/Driver/Folio.php]: Yes, this is better. Done.

**Review by demiankatz — APPROVED** [review]

**Comment by demiankatz** [comment]
Thanks! Looks good to me. I'll ping the Slack channel to see if anyone else has time for a double-check before I merge.

**Comment by ihardy** [comment]
Thanks @bbusenius and @demiankatz sorry about the slow response. This looks good to me. Tried it out on one of our test instances and I see the new notes data coming through etc.

**Comment by demiankatz** [comment]
Thanks, @ihardy! Merging now.

**Comment by bbusenius** [comment]
Thanks @ihardy and @demiankatz!

**Pull request merged**
